src/utils/hydro_utilities.h

```diff
@@ -63,11 +63,42 @@ inline __host__ __device__ Real Calc_Temp(Real const &P, Real const &n)
   return T;
 }
 
+/*!
+ * \brief Compute the temperature from the conserved variables
+ *
+ * \param[in] E The energy
+ * \param[in] d The density
+ * \param[in] mx The momentum in the X-direction
+ * \param[in] my The momentum in the Y-direction
+ * \param[in] mz The momentum in the Z-direction
+ * \param[in] gamma The adiabatic index
+ * \param[in] n The number density
+ * \param[in] magnetic_x The cell centered magnetic field in the X-direction
+ * \param[in] magnetic_y The cell centered magnetic field in the Y-direction
+ * \param[in] magnetic_z The cell centered magnetic field in the Z-direction
+ * \return Real The temperature of the gas in a cell
+ */
+inline __host__ __device__ Real Calc_Temp_Conserved(Real const E, Real const d, Real const mx, Real const my,
+                                                    Real const mz, Real const gamma, Real const n,
+                                                    Real const magnetic_x = 0.0, Real const magnetic_y = 0.0,
+                                                    Real const magnetic_z = 0.0)
+{
+  Real const P = Calc_Pressure_Conserved(E, d, mx, my, mz, gamma, magnetic_x, magnetic_y, magnetic_z);
+  return Calc_Temp(P, n);
+}
+
 #ifdef DE
-inline __host__ __device__ Real Calc_Temp_DE(Real const &d, Real const &ge, Real const &gamma, Real const &n)
+/*!
+ * \brief Compute the temperature when DE is turned on
+ *
+ * \param total_gas_energy The total gas energy in the cell. This is the value stored in the grid
```

Comment:
Might be clearer to specifically say the name of the grid_enum index (as in grid_enum::Energy)

Reply:
done

Reply:
Also, I'm not sure that total_gas_energy is the clearest variable name. [In the style guide](https://github.com/cholla-hydro/cholla/discussions/206#discussioncomment-4429288), I think we decided on energy_tot to refer to the total gas energy (unless I'm misunderstanding something)

Reply:
I don't think we want energy_tot, since that would imply the total energy, i.e. thermal + kinetic + magnetic etc. etc. My reading of the style guide is that it should be total_gas_energy.

Reply:
My impression is that the total is not actually necessary, since we usually mean the total when we say any of these (energy, kinetic energy, etc.) and really what we should be doing is clarifying when we are using the specific version (i.e. the old ge should be specific_gas_energy).

Reply:
> I don't think we want energy_tot, since that would imply the total energy, i.e. thermal + kinetic + magnetic etc. etc. My reading of the style guide is that it should be total_gas_energy.
It is `total_gas_energy` I'm confused

> My impression is that the total is not actually necessary...
Sure, but I'd rather be extra clear since the ambiguity is what led to this multi-hour debacle.

Reply:
In fact, I don't see energy_total anywhere in the code, and would advocate for this approach. I think gas_energy is consistent with the way this is labeled elsewhere (i.e. in reconstruction.h).

Reply:
Sorry, I think I created a bit of confusion because I was still editing my comments before I realized that you guys were already replying. However, I agree that the `total_` isn't that clear. I think the clearest thing to do is clarify in the doc string and follow what's done elsewhere in the code by calling it `gas_energy`.

Reply:
I've changed the variable name to `gas_energy`.